packages/workflow/src/documents.ts

```diff
@@ -70,57 +70,70 @@ export async function uploadCertificateAttachmentsToDocumentsStore<
+  const parsedUrl = new URL(url)
+  const filePath = parsedUrl.pathname
+
+  return filePath
+}
+
+function uploadOrNormaliseSignatureData(
+  signature: string,
+  authHeader: IAuthHeader
+) {
+  if (isBase64FileString(signature)) {
+    return uploadBase64ToMinio(signature, authHeader)
+  }
+
+  if (isPresignedUrl(signature)) {
+    return getMinioUri(signature)
+  }
+  return signature
+}
+
 export async function uploadBase64AttachmentsToDocumentsStore(
   record: BirthRegistration | DeathRegistration | MarriageRegistration,
   authHeader: IAuthHeader
 ) {
   /*
    * @todo input schema and target all AttachmentInput types automatically
    */
-  if (
-    record.registration?.informantsSignature &&
-    isBase64FileString(record.registration?.informantsSignature)
-  ) {
-    record.registration.informantsSignature = await uploadBase64ToMinio(
-      record.registration.informantsSignature,
-      authHeader
-    )
+  if (record.registration?.informantsSignature) {
+    record.registration.informantsSignature =
+      await uploadOrNormaliseSignatureData(
+        record.registration.informantsSignature,
+        authHeader
+      )
   }
-  if (
-    record.registration?.groomSignature &&
-    isBase64FileString(record.registration?.groomSignature)
-  ) {
-    record.registration.groomSignature = await uploadBase64ToMinio(
+  if (record.registration?.groomSignature) {
+    record.registration.groomSignature = await uploadOrNormaliseSignatureData(
       record.registration.groomSignature,
       authHeader
     )
   }
-  if (
-    record.registration?.brideSignature &&
-    isBase64FileString(record.registration?.brideSignature)
-  ) {
-    record.registration.brideSignature = await uploadBase64ToMinio(
+  if (record.registration?.brideSignature) {
+    record.registration.brideSignature = await uploadOrNormaliseSignatureData(
       record.registration.brideSignature,
       authHeader
     )
   }
-  if (
-    record.registration?.witnessOneSignature &&
-    isBase64FileString(record.registration?.witnessOneSignature)
-  ) {
-    record.registration.witnessOneSignature = await uploadBase64ToMinio(
-      record.registration.witnessOneSignature,
-      authHeader
-    )
+  if (record.registration?.witnessOneSignature) {
+    record.registration.witnessOneSignature =
+      await uploadOrNormaliseSignatureData(
+        record.registration.witnessOneSignature,
+        authHeader
+      )
   }
-  if (
-    record.registration?.witnessTwoSignature &&
-    isBase64FileString(record.registration?.witnessTwoSignature)
-  ) {
-    record.registration.witnessTwoSignature = await uploadBase64ToMinio(
-      record.registration.witnessTwoSignature,
-      authHeader
-    )
+  if (record.registration?.witnessTwoSignature) {
+    record.registration.witnessTwoSignature =
+      await uploadOrNormaliseSignatureData(
+        record.registration.witnessTwoSignature,
+        authHeader
+      )
   }
   if (record.registration?.attachments) {
```

Comment:
should we be doing this for the rest of the attachments as well?

Reply:
I think so. Ideally, uploading attachments would already be done in gateway so workflow would not have to think about this at all. What do you think Tameem?

Reply:
Yep. Also we don't really support updating the `documents` currently, I think. So it wouldn't matter much to do the same for attachment files.

Reply:
@Nil20 we probably should do this for the attachments too because we can just remove a document and upload a separate one in it's place. Let's also ask Syeda if it's indeed a valid case or not

Reply:
For attachments, both `presignedUrl` and `uri` are handled when we create the payload in client.
https://github.com/opencrvs/opencrvs-core/blob/develop/packages/client/src/forms/register/mappings/mutation/field-mappings.ts#L519